### PR TITLE
Add e2e test for GCS

### PR DESCRIPTION
**Pull request by khrm**
This adds an e2e test for GCS using storage emulator by passing env STORAGE_EMULATOR_HOST. Reference:
https://cloud.google.com/go/docs/reference/cloud.google.com/go/storage/latest

### CODE REVIEW

**Review by khrm — COMMENTED** [review]
We need to merge #84 first.

**Comment by khrm** [comment]
@vdemeester Should I write this in Go or is bash fine?
It will be just calling a PR in GO and checking as we do in the Tekton repo but it will needlessly pull pipeline repo.

File: tests/e2e.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    exit 0
+else
+    echo "Error: Pipelinerun failed or timed out"
+    kubectl get pipelinerun
+    kubectl describe pipelinerun
+    kubectl get taskrun
+    kubectl describe taskrun
+    exit 1
+fi
+
```
Comment (PuneetPunamiya) [inline tests/e2e.sh:39]: I didn't get what are we asserting here ?
If I'm not wrong we are just creating GCS emulateor deployment and installing fetch and upload stepActions and finally creating pipelineruns

Comment (khrm) [reply tests/e2e.sh:39]: We are checking whether pipelinerun succeeded or not.

Comment (khrm) [reply tests/e2e.sh:48]: And also in the last step, we are verifying whether cache was uploaded and fetched or not. It has different workspace where we fetch the cache.

File: tests/test-pipelineruns/test-pipelinerun-gcs.yaml [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+                  value: $(workspaces.test.path)/cache
+                - name: workingdir
+                  value: $(workspaces.source.path)/repo
+                - name: awsCredentialFile
+                  value: $(workspaces.cred.path)/credentials
+                - name: awsConfigFile
+                  value: $(workspaces.cred.path)/config
+                - name: googleCredentialsPath
+                  value: $(workspaces.cred.path)/creds.json
+            - name: validate-cache
```
Comment (khrm) [inline tests/test-pipelineruns/test-pipelinerun-gcs.yaml:143]: @PuneetPunamiya See this. In this step, we validate cache. In the previous step, we fetched catch in a different workspace - `test`.

**Review by vdemeester — COMMENTED** [review]
I know it's probably easier to write it in bash, but why don't we have all our tests in `go` instead (under the `e2e` build tag) ?

**Comment by khrm** [comment]
@vdemeester Is it fine if I pull pipelines repo to write tests?

**Comment by vdemeester** [comment]
@khrm it depends in what you pull it for 🤭 if it is for 2 functions we can duplicate them here 🤭

**Comment by khrm** [comment]
@vdemeester Basically, we would need the pipelinerun client to check status. In script, I am using kubectl in bash.

**Comment by vdemeester** [comment]
> @vdemeester Basically, we would need the pipelinerun client to check status. In script, I am using kubectl in bash.

ah yes, it is fine 👍🏼

File: Makefile [hunk 3]
```diff
@@ -6,8 +6,10 @@ TIMEOUT_UNIT = 20m
 GOFUMPT=gofumpt
 
 e2e-coverage: ## run e2e tests with coverage
-	@go test -failfast -count=1 -tags=$(E2E_TAG) ./tests -coverpkg=./... -coverprofile /tmp/coverage.out
+	tests/e2e.sh
```
Comment (vdemeester) [inline Makefile:9]: Might want to rename this "e2e-prepare.sh" or something but it could be done in a follow-up.

**Review by vdemeester — APPROVED** [review]

**Pull request merged**